t/test.pl

```diff
@@ -1743,6 +1743,16 @@ ($;$)
     # create the watchdog.
     if ($timeout == 0) {
         if ($watchdog_thread) {
+
+            # Don't kill the watchdog until it has set up the signal handler.
+            # It tells us that it has done so by detaching itself immediately
+            # afterwards.  (The 'is_running()' is a fail-safe.)
+            while (     $watchdog_thread->is_running()
+                   && ! $watchdog_thread->is_detached())
+            {
+                sleep 1;
```

Comment:
Is that a long time? sleep 0.1 would be more reactive, and wouldnt burden the system that much more in the grand scheme of things.

Reply:
I didn't realize that a non-integer was possible.  But I tried it, and it didn't work, and then I looked at perlfunc, and it takes an integer argument.  We don't probably want to load some module in t/test.pl that could sleep less than a second.

Reply:
Yeah, my bad. That would need Time::HiRes, and of course we cant load that in this code. Sorry for the noise.